### DISCUSSION TITLE
Numercial difficulties for surface form: differential

### DISCUSSION THREAD

**Discussion by dion-w** [pybamm-team/PyBaMM#2468] [General]
Hello everyone,
i noticed something peculiar lately. I normally do huge parameter studies for particle size and porosity in the standard DFN. This is with constant current or constant power. As i am comparing these parameter studies with the operation mode of a driving cycle i'm using the submodel "surface form: differential" for all operation modes. This submodel introduces a double layer capacity in our system and therefore the time derivative of the potential at the interface is introduced as an ordinary differential equation. I do this to model the dynamic behaviour of my system closer to reality. Also i am using the casadi solver in safe mode. 
What i noticed that out of 7500 simulations, most of the time around 100 simulations don't make it, sometimes up to a thousand more, which means: At some point in the solving process the solver won't produce anymore timesteps or is hung up in an interval of solving the system. Even if i give a simulation 24 hours of runtime, no solution will be created. The calculation just continues. The solver doesn't even recognize that the timestep for example may be too small. This only happens for the double layer model with "surface form differential" though. If i omit the double layer everything just solves fine and very fast for all 7500 simulations.
I think this behvaiour is quite odd. A double layer is needed for operating cells dynamically and especially for calculation of impedance spectra this is definitely a useful feature. Has anybody ever encountered such numerical difficulties?

**Comment by valentinsulzer** [comment]
Yeah, we're aware of this issue but don't know how to fix it at the moment. There is an issue open (#2173 ) so let's continue discussion there
